File: Java Based Archives/DI&Dagger/app/src/main/java/com/android/di/views/MainActivity.java
package com.android.di.views;


import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.android.di.App;
import com.android.di.Battery;
import com.android.di.R;
import com.android.di.SmartPhone;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {
    @Inject
    SmartPhone smartPhone;
    @Inject
    Battery battery;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        /*smartPhone = smartPhoneComponent.getSmartPhone();*/
        App.getApp().getSmartPhoneComponent().inject(this);
        smartPhone.makeACall();

        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, MainActivity2.class);
                startActivity(intent);


            }
        });

    }
}

File: Views/DependencyInjection/app/src/main/java/com/example/dependencyInjection/MainActivity.kt
package com.example.dependencyInjection

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import javax.inject.Inject

class MainActivity : AppCompatActivity() {
    @Inject lateinit var smartPhone: SmartPhone

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        (application as SmartPhoneApplication). smartPhoneFunctionComponent
          .inject(this)
        smartPhone.makeACallWithRecording()


        /**val smartPhone = SmartPhone(
            Battery(),
            SIMCard(ServiceProvider()),
            MemoryCard()
        )
            .makeACallWithRecording()
         */


    }
}
